### PR TITLE
[FIX] mail: inbox notif from chatter should not play sound

### PR DESCRIPTION
**Pull request by alexkuhn**
Follow-up of https://github.com/odoo/odoo/pull/182347

PR above improved out-of-focus counter, one of such improvements is to take new inbox messages into account.

However, doing so had to unintended side-effect to make them play sound, open chat bubble, and display more than 1 notification.

This happens because the inbox notification were treated like new message in chat notifications.

This commit fixes the issue by limiting PR above of inbox new messages to contribute for the out-of-focus counter. In other words, the following behaviors have been fixed:
- new inbox messages from chatter no longer open a chat bubble
- new inbox messages from chatter no longer play sound
- new inbox messages from chatter only show 1 notification at most

### CODE REVIEW

File: addons/mail/static/src/core/common/out_of_focus_service.js [hunk 1]
```diff
@@ -57,6 +57,9 @@ export class OutOfFocusService {
             title: notificationTitle,
             type: "info",
         });
+        if (thread?.model === "discuss.channel") {
```
Comment (zel-odoo) [inline addons/mail/static/src/core/common/out_of_focus_service.js:60]: `thread` can be empty?

File: addons/mail/static/src/core/common/out_of_focus_service.js [hunk 2]
```diff
@@ -57,6 +57,9 @@ export class OutOfFocusService {
...
+            this._playSound();
```
Comment (zel-odoo) [inline addons/mail/static/src/core/common/out_of_focus_service.js:61]: I would still prefer to send sound inside the `sendOdooNotification` and `sendNativeNotification`, because `sendNativeNotification` can fail and seems to be off when having the sound but actually not getting notified

File: addons/mail/static/src/core/web/mail_core_web_service.js [hunk 3]
```diff
@@ -41,7 +41,7 @@ export class MailCoreWeb {
             if (message.thread && notifId > message.thread.message_needaction_counter_bus_id) {
                 message.thread.message_needaction_counter++;
             }
-            message.thread?.notifyMessageToUser(message);
+            this.store.env.services["mail.out_of_focus"].notify(message, this);
```
Comment (zel-odoo) [inline addons/mail/static/src/core/web/mail_core_web_service.js:44]: nit-pick: I think the precious one was better and more readable and had a clearer API.

Comment (alexkuhn) [reply addons/mail/static/src/core/web/mail_core_web_service.js:44]: `notifyMessageToUser` targets a thread, this means the message has to be linked to a thread.
Inbox messages are not necessarily linked to a thread, e.g. `You have been assigned to` messages.
We should also put them as contribution to the out-of-focus counter, as they are inbox messages.

The `this` is wrong as 2nd parameter, I'll fix it!

File: addons/mail/static/src/core/common/out_of_focus_service.js (hunk 2 above)
Comment (alexkuhn) [reply addons/mail/static/src/core/common/out_of_focus_service.js:61]: If the `sendNativeNotiication` fails, we do `sendOdooNotification`, so this will necessarily result to `_playSound()`.

Moving it to `sendNotification()` makes diffing these LOCs once, which is best because I put a conditional to affect only `discuss.channel` messages.

File: addons/mail/static/src/core/common/out_of_focus_service.js (hunk 1 above)
Comment (alexkuhn) [reply addons/mail/static/src/core/common/out_of_focus_service.js]: Yes, the thread is optional. Actually this is always `message.thread`, so I made more diff to make 2nd parameter `thread` deprecated.

File: addons/mail/static/src/core/common/out_of_focus_service.js (hunk 2 above)
Comment (seb-odoo) [reply addons/mail/static/src/core/common/out_of_focus_service.js:61]: This is a lot of API change in stable though, can't we have a play sound param to the other methods, which is true by default to keep the behavior, and only passed false from where we want to prevent it?

File: addons/mail/static/src/core/common/out_of_focus_service.js [hunk 4]
```diff
@@ -57,6 +57,9 @@ export class OutOfFocusService {
             title: notificationTitle,
             type: "info",
         });
+        if (message.thread?.model === "discuss.channel") {
```
Comment (seb-odoo) [inline addons/mail/static/src/core/common/out_of_focus_service.js:60]: Conditions about channel should be move to discuss folder as much as possible.

File: addons/mail/static/src/core/common/out_of_focus_service.js (hunk 2 above)
Comment (zel-odoo) [reply addons/mail/static/src/core/common/out_of_focus_service.js:61]: ok i see what you mean.
Can we change the API of `async notify(message, channel)` to `async notify(message, channel, , sound = false)` I don't think that disabling/enabling something based on a hard-coded check of the model type inside of a basic function like `notify` is not a good idea and it is harder to maintain the code.

For example, I want to disable the sound when I trying to implement the mute stuff and found out it was coupled with another feature that causes my channel to always have sound. It can be really annoying.

Comment (zel-odoo) [reply addons/mail/static/src/core/common/out_of_focus_service.js:61]: > This is a lot of API change in stable though, can't we have a play sound param to the other methods, which is true by default to keep the behavior, and only passed false from where we want to prevent it?

you were faster than me! xd

File: addons/mail/static/src/core/common/out_of_focus_service.js [hunk 5]
```diff
@@ -109,25 +113,29 @@ export class OutOfFocusService {
...
         const notification = new Notification(title, {
             body: message,
             icon: "/mail/static/src/img/odoobot_transparent.png",
         });
         notification.addEventListener("click", ({ target: notification }) => {
             window.focus();
             notification.close();
+            if (sound) {
+                this._playSound();
+            }
```
Comment (seb-odoo) [inline addons/mail/static/src/core/common/out_of_focus_service.js:137]: Why is this moved inside the handler?

File: addons/mail/static/src/core/common/out_of_focus_service.js [hunk 6]
```diff
@@ -56,7 +56,11 @@ export class OutOfFocusService {
             message: notificationContent,
             title: notificationTitle,
             type: "info",
+            sound: false,
         });
+        if (message.thread?.model === "discuss.channel") {
+            this._playSound();
```
Comment (seb-odoo) [inline addons/mail/static/src/core/common/out_of_focus_service.js:62]: Why is there an extra play sound here?

File: addons/mail/static/src/core/common/out_of_focus_service.js (hunk 5 above)
Comment (alexkuhn) [reply addons/mail/static/src/core/common/out_of_focus_service.js:134]: oops

**Comment by alexkuhn** [comment]
@robodoo r+
